sympy/sets/sets.py

```diff
@@ -217,17 +217,18 @@ def _complement(self, other):
             return S.EmptySet
 
         elif isinstance(other, FiniteSet):
-            from sympy.utilities.iterables import sift
+            from sympy.core.logic import fuzzy_bool
 
             def ternary_sift(el):
                 contains = self.contains(el)
-                return contains if contains in [True, False] else None
+                return fuzzy_bool(contains)
```

Comment:
You can create and pass a lambda function like `lambda x: fuzzy_bool(self.contains(x))`, but it's OK.